atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -2129,53 +2126,60 @@ public void compactInternally(TableReference tableRef) {
                 }
             } catch (Exception e) {
                 numberOfUnreachableNodes++;
             }
         }
-        int quorumSize = replicationFactor / 2 + 1;
-        if (numberOfUnreachableNodes > (replicationFactor - quorumSize)) {
-            return NodeAvailabilityStatus.NO_QUORUM_AVAILABLE;
-        } else if (numberOfUnreachableNodes == 0) {
-            return NodeAvailabilityStatus.ALL_AVAILABLE;
-        } else {
-            return NodeAvailabilityStatus.QUORUM_AVAILABLE;
-        }
+        return getClusterAvailabilityStatus(numberOfUnreachableNodes);
     }
 
-    private void readAtSpecifiedConsistency(ConsistencyLevel consistency) throws TException {
-        clientPool.run(client -> {
-            String lockRowName = getHexEncodedBytes(CassandraConstants.GLOBAL_DDL_LOCK_ROW_NAME);
-            String lockColName = getHexEncodedBytes(CassandraConstants.GLOBAL_DDL_LOCK_COLUMN_NAME);
-            String selectCql = String.format(
-                    "SELECT \"value\" FROM \"%s\" WHERE key = %s AND column1 = %s AND column2 = -1;",
-                    schemaMutationLockTable.getOnlyTable().getQualifiedName(),
-                    lockRowName,
-                    lockColName);
-            return runCqlQuery(selectCql, client, consistency);
+    private boolean doesConfigReplicationFactorMatchWithCluster() {
+        return clientPool.run(client -> {
+            try {
+                CassandraVerifier.currentRfOnKeyspaceMatchesDesiredRf(client, configManager.getConfig(), false);
+                return true;
+            } catch (Exception e) {
+                return false;
+            }
         });
     }
 
-    private static CqlResult runCqlQuery(String query, Cassandra.Client client, ConsistencyLevel consistency)
-            throws TException {
-        ByteBuffer queryBuffer = ByteBuffer.wrap(query.getBytes(StandardCharsets.UTF_8));
-        return client.execute_cql3_query(queryBuffer, Compression.NONE, consistency);
+    private boolean partitionerIsValid(InetSocketAddress host) {
+        try {
+            clientPool.runOnHost(host, clientPool.validatePartitioner);
+            return true;
+        } catch (Exception e) {
+            return false;
+        }
     }
 
-    private static String getHexEncodedBytes(String str) {
-        return CassandraKeyValueServices.encodeAsHex(str.getBytes(StandardCharsets.UTF_8));
+    private NodeAvailabilityStatus getClusterAvailabilityStatus(int numberOfUnreachableNodes) {
+        if (numberOfUnreachableNodes == 0) {
+            return NodeAvailabilityStatus.ALL_AVAILABLE;
+        } else if (fewerThanRfByTwoNodesUnreachable(numberOfUnreachableNodes)) {
+            return NodeAvailabilityStatus.QUORUM_AVAILABLE;
+        } else {
+            return NodeAvailabilityStatus.NO_QUORUM_AVAILABLE;
+        }
+    }
+
+    private boolean fewerThanRfByTwoNodesUnreachable(int numberOfUnreachableNodes) {
```

Comment:
This method name describes perfectly _what_ the method does, but not _why_. The "what" can be readily understood by reading the code, so how about `isQuorumAvailable`?